string name
---
string[] par_names
string[] par_values
bool success